tensordict/tensordict.py

```diff
@@ -4910,6 +4929,81 @@ def _cat(
         return out
 
 
+@implements_for_lazy_td(torch.cat)
+def _lazy_cat(
+    list_of_tensordicts: Sequence[LazyStackedTensorDict],
+    dim: int = 0,
+    out: LazyStackedTensorDict | None = None,
+) -> LazyStackedTensorDict:
+    if not list_of_tensordicts:
+        raise RuntimeError("list_of_tensordicts cannot be empty")
+
+    batch_size = list(list_of_tensordicts[0].batch_size)
+    if dim < 0:
+        dim = len(batch_size) + dim
+    if dim >= len(batch_size):
+        raise RuntimeError(
+            f"dim must be in the range 0 <= dim < len(batch_size), got dim"
+            f"={dim} and batch_size={batch_size}"
+        )
+    stack_dim = list_of_tensordicts[0].stack_dim
+    if any((td.stack_dim != stack_dim) for td in list_of_tensordicts):
+        raise RuntimeError("cat lazy stacked tds must have same stack dim")
+
+    batch_size[dim] = sum(td.batch_size[dim] for td in list_of_tensordicts)
+    batch_size = torch.Size(batch_size)
+
+    new_dim = dim
+    if dim > stack_dim:
+        new_dim = dim - 1
+
+    if out is None:
+        out = []
+        if dim == stack_dim:  # if dim is stack, just add all to same list
+            for lazy_td in list_of_tensordicts:
+                out += lazy_td.tensordicts
+        else:
+            for i in range(len(list_of_tensordicts[0].tensordicts)):
+                out.append(
+                    torch.cat(
+                        [lazy_td.tensordicts[i] for lazy_td in list_of_tensordicts],
+                        new_dim,
+                    )
+                )
+        return LazyStackedTensorDict(*out, stack_dim=stack_dim)
+    else:
+        if out.batch_size != batch_size:
+            raise RuntimeError(
+                "out.batch_size and cat batch size must match, "
+                f"got out.batch_size={out.batch_size} and batch_size"
+                f"={batch_size}"
+            )
+        if out.stack_dim != stack_dim:
+            raise RuntimeError(
+                "out.stack_dim and elements stack_sim did not match, "
+                f"got out.stack_dim={out.stack_dim} and stack_dim={stack_dim}"
+            )
```

Comment:
Is this technically impossible or too much trouble?

Reply:
i am not sure. 
I have difficulty picturing what it would look like, what do you think?

Reply:
let me think about that

Reply:
I am not understanding this part
```
dim = 0
for i, sub_dest in enumerate(td1.tensordicts):
    index = (slice(None),) * td1.stack_dim + (i,)
    tds_to_cat = [_td[index] for _td in tds]
    torch.cat(tds_to_cat, dim, out=sub_dest)
```

here at the end you are trying to cat lazy stacks onto a tensordict (non-lazy), how is this supposed to work?

Reply:
I'm indexing `td1` along its stack-dim, so I get a regular td if it's a stack of regular tds. That gives me `tds_to_cat`
Since I enumerate all the tds in td1 to get `sub_dest` I get the same dim reduction in both

Reply:
`tds_to cat` is a list of lazystackedtds while `sub_dest` is a regular td. 
Is that cat operation supposed to handle that?

Reply:
in other words, you are asking the lazy cat to work with a non lazy output, seems impossible to me